pennylane_qiskit/converter.py

```diff
@@ -160,8 +170,24 @@ def _function(params: dict = None, wires: list = None):
             operation_wires = [wire_map[(qubit.register.name, qubit.index)] for qubit in op[1]]
 
             if instruction_name in inv_map and inv_map[instruction_name] in pennylane_ops.ops:
+                # Extract the bound parameters from the operation. If the bound parameters are a
+                # Qiskit ParameterExpression, then replace it with the corresponding PennyLane
+                # variable from the var_ref_map dictionary.
+
+                parameters = []
+                for p in op[0].params:
+
+                    _check_parameter_bound(p, var_ref_map)
+
+                    if isinstance(p, ParameterExpression):
+                        if p.parameters:
+                            parameters.append(var_ref_map.get(min(p.parameters)))
```

Comment:
Not too sure about this line here, in particular `min(p.parameters)`. How come such a call is made? Perhaps naming a certain part of this line (e.g. `min_param=min(p.parameters)`) and changing this line accordingly could help with providing further clues to the reader :)

Reply:
Does `len(p.parameters) == 1`? If not, what happens to `p`'s other parameters?

Reply:
Oh, this is partly because

* Python doesn't provide a way to index into a set, and

* Due to a PennyLane restriction, the Qiskit ParameterExpression must always be a set containing a single parameter.

As a result, `min(single_element_set)` is a useful way of extracting the element. Another approach is `single_element_set.pop()`, but I tend not to favour `pop` due to its side effects.

Side note: probably out of scope for this PR, but we should create a PR that raises an error if `len(p.parameters) > 1`.

Reply:
Oh, got ya! So a [`ParameterExpression` contains](https://qiskit.org/documentation/_modules/qiskit/circuit/parameterexpression.html) a 

> Mapping of Parameter instances to the sympy.Symbol serving as their placeholder in expr.

Having seen that, I was also wondering why simply a `Parameter` could not do it here, but this now allows for `ParameterExpression` instances that contain only one `Parameter` as well.

Reply:
Yep - the issue seems to be that even if you start off with a single `Parameter`, after binding Qiskit converts _everything_ into a `ParameterExpression`. So `Parameter` objects never appear at this point in the logic.